src/XCCDF_POLICY/xccdf_policy_remediate.c

```diff
@@ -493,39 +499,58 @@ int xccdf_policy_rule_result_remediate(struct xccdf_policy *policy, struct xccdf
 	if (xccdf_rule_result_get_result(rr) != XCCDF_RESULT_FAIL)
 		return 0;
 
+	// if a miscellaneous error happens (fix unsuitable or if we want to skip it for any reason
+	// we set misc_error to one, and the fix will be reported as error (and not skipped without log like before)
+	int misc_error=0;
+
 	if (fix == NULL) {
 		fix = _find_suitable_fix(policy, rr);
-		if (fix == NULL)
-			// We may want to append xccdf:message about missing fix.
-			return 0;
+		if (fix == NULL) {
+			// We want to append xccdf:message about missing fix.
+			_rule_add_info_message(rr, "No suitable fix found.");
+			xccdf_rule_result_set_result(rr, XCCDF_RESULT_ERROR);
```

Comment:
This is controversial.

The first point of view is that if you have no remediation nothing changed so the rule result is still fail.

The second point of view is that if users run "oscap xccdf remediate" users expect that the rule will start passing after the command finishes and if it doesn't it should be an error.

If you have found anything about that in the XCCDF specification, please point me to that, but I haven't found anything. To be on the safe side I prefer to keep the current behavior, which is to keep the XCCDF_RESULT_FAIL.

Reply:
I understand this point of view too :) that's done for both change request.